UPGRADE.rst

```diff
@@ -85,6 +85,19 @@ for example:
      wget https://packages.matrix.org/debian/pool/main/m/matrix-synapse-py3/matrix-synapse-py3_1.3.0+stretch1_amd64.deb
      dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
 
+Upgrading to v1.34.0
+====================
+
+Deprecation of the built-in account validity feature
+----------------------------------------------------
+
+As of Synapse v1.34.0, the built-in account validity feature is deprecated in favour of
+plugin modules. If you are using this feature and want to keep its current behaviour,
+please use the `synapse-email-account-validity <https://github.com/matrix-org/synapse-email-account-validity>`_
+module.
+
+Synapse's built-in account validity feature will be removed in a future release.
```

Comment:
Should we define this more explicitly?

(E.g., giving people three months to migrate would mean dropping in Synapse 1.40 on 2021-08-10)

Reply:
Agreed that it needs defining more precisely. Given the time this PR is taking to get through, I'll determine that once the review is done since we'll at least know which version this change will be shipped in.

Reply:
historically we've always ended up looking stupid when we say "we will do X on date Y", and Y then sails past. It might be possible to come up with some wording that allows us wiggle room.